pkg/planner/core/tiflash_selection_late_materialization.go

```diff
@@ -278,11 +278,9 @@ func predicatePushDownToTableScanImpl(sctx base.PlanContext, physicalSelection *
 
 // PushedDown is used to push down the selected conditions from PhysicalSelection to PhysicalTableScan.
 // Used in unit test, so it is exported.
-func PushedDown(sel *PhysicalSelection, ts *PhysicalTableScan, selectedConds []expression.Expression, selectedSelectivity float64) {
+func PushedDown(sel *PhysicalSelection, ts *PhysicalTableScan, selectedConds []expression.Expression, _ float64) {
 	// remove the pushed down conditions from selection
 	removeSpecificExprsFromSelection(sel, selectedConds)
 	// add the pushed down conditions to table scan
 	ts.LateMaterializationFilterCondition = selectedConds
-	// Update the row count of table scan after pushing down the conditions.
-	ts.StatsInfo().RowCount *= selectedSelectivity
```

Comment:
Although, the post optimizer is the step after cost model which means the row count and cost will not affect the best plan, the row count should be keep the actual value instead of ignore the change of row count and cost.